### PR TITLE
Fix accept and return to end with empty input

### PR DESCRIPTION
**Pull request by jannyHou**
Add empty check to `validName` breaks remote-method generator, since in accept and return prompt, it checks validation before checking empty. 
I change the logic in remote-method generator so empty input could pass.

### CODE REVIEW

**Comment by jannyHou** [comment]
@bajtos Please review :) thanks

**Comment by bajtos** [comment]
@jannyHou @0candy are we sure only remote-method generator is affected?

**Comment by 0candy** [comment]
I think anywhere that uses validateName and allows `enter empty name when done` will fail. But I think @jannyHou said some places check for empty and exit before validating.

**Comment by bajtos** [comment]
Makes sense.

We discussed this on Slack with Janny, agreed to refactor the code a bit: rename `validateName` to `validateRequiredName`, add new method `validateOptionalName` and then use these two methods as appropriate. Hopefully it will make it more clear which places allow empty values and which do not.

**Comment by jannyHou** [comment]
@bajtos `validateRequiredName` and `validateOptionalName` are applied. Please review:)

**Comment by jannyHou** [comment]
@slnode test please

File: lib/helpers.js [hunk 1]
```diff
@@ -60,14 +60,27 @@ exports.checkPropertyName = function(name) {
...
  * @returns {String|Boolean}
  */
-exports.validateName = function (name) {
+exports.validateRequiredName = function (name) {
+  var result = exports.validateOptionalName(name);
+  if (result !== true) return result;
   if (!name) {
     return 'Name is required';
   }
+  return true;
```
Comment (bajtos) [inline lib/helpers.js]: How about simplifying the method by checking whether the name is truthy first?

```js
if (!name) {
  return 'Name is required';
}

return exports.validateOptionalName(name);
```

**Comment by bajtos** [comment]
One more comment (see above). The rest LGTM, no need for further reviews.

**Comment by bajtos** [comment]
> generator-loopback/node=4.x,os=amazon — Failed! (9a81230)

Please check why the build is failing on CI. We should at least get generator-loopback builds green, if not all dependants.

**Comment by jannyHou** [comment]
@slnode test please

File: lib/helpers.js (hunk 1 above)
Comment (jannyHou) [inline lib/helpers.js]: agreed:)

**Pull request merged**